src/test/java/mate/academy/service/BookServiceTest.java

```diff
@@ -0,0 +1,307 @@
+package mate.academy.service;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dto.book.BookDto;
+import mate.academy.dto.book.BookDtoWithoutCategoryIds;
+import mate.academy.dto.book.BookSearchParametersDto;
+import mate.academy.dto.book.CreateBookRequestDto;
+import mate.academy.exception.EntityNotFoundException;
+import mate.academy.mapper.BookMapper;
+import mate.academy.model.Book;
+import mate.academy.repository.book.BookRepository;
+import mate.academy.repository.book.spec.BookSpecificationBuilder;
+import mate.academy.service.impl.BookServiceImpl;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+
+@ExtendWith(MockitoExtension.class)
+public class BookServiceTest {
+    @InjectMocks
+    private BookServiceImpl bookService;
+    @Mock
+    private BookRepository bookRepository;
+    @Mock
+    private BookMapper bookMapper;
+    @Mock
+    private BookSpecificationBuilder bookSpecificationBuilder;
+
+    @Test
+    @DisplayName("Verify save() method works")
+    public void save_ValidCreateBookRequestDto_ReturnSavedBook() {
+        // Given
+        CreateBookRequestDto requestDto = new CreateBookRequestDto();
+        requestDto.setTitle("Test Title");
+        requestDto.setAuthor("Test Author");
+        requestDto.setIsbn("9781122334455");
+        requestDto.setPrice(BigDecimal.valueOf(100));
+
+        Book book = new Book();
+        book.setTitle(requestDto.getTitle());
+        book.setAuthor(requestDto.getAuthor());
+        book.setIsbn(requestDto.getIsbn());
+        book.setPrice(requestDto.getPrice());
+
+        BookDto bookDto = new BookDto();
+        bookDto.setId(1L);
+        bookDto.setTitle(book.getTitle());
+        bookDto.setAuthor(book.getAuthor());
+        bookDto.setIsbn(book.getIsbn());
+        bookDto.setPrice(book.getPrice());
+
+        when(bookMapper.toEntity(requestDto)).thenReturn(book);
+        when(bookRepository.save(book)).thenReturn(book);
+        when(bookMapper.toDto(book)).thenReturn(bookDto);
+
+        // When
+        BookDto savedBookDto = bookService.save(requestDto);
+
+        // Then
+        Assertions.assertThat(savedBookDto).isEqualTo(bookDto);
+
+        verify(bookMapper, Mockito.times(1)).toEntity(requestDto);

```

Comment:
times(1) is by default
remove and check other places
```suggestion
        verify(bookMapper).toEntity(requestDto);
```